### ISSUE TITLE
Very misleading diagnostic for template `>>` in C++03

### ISSUE THREAD

**Issue by cjdb** [llvm/llvm-project#68454]
The diagnostic for the code below is very misleading, and doesn't help diagnose the problem.

**Source:**
```cpp
template<class T, int N>
inline static const bool sized = sizeof(T) == N;

template<bool B = sized<int, 4>>
int f()
{
    return B ? 0 : 1;
}

int main()
{
    return f();
}
```

**Diagnostic:**
```
<source>:2:26: warning: variable templates are a C++14 extension [-Wc++14-extensions]
    2 | inline static const bool sized = sizeof(T) == N;
      |                          ^
<source>:2:1: warning: inline variables are a C++17 extension [-Wc++17-extensions]
    2 | inline static const bool sized = sizeof(T) == N;
      | ^
<source>:5:5: error: expected '(' for function-style cast or type construction
    5 | int f()
      | ~~~ ^
<source>:13:2: error: expected '>'
   13 | }
      |  ^
<source>:4:24: note: to match this '<'
    4 | template<bool B = sized<int, 4>>
      |                        ^
<source>:13:2: error: expected ',' or '>' in template-parameter-list
   13 | }
      |  ^
<source>:13:2: error: expected unqualified-id
```

**Resolution:** Change `4>>` to `4> >`

**Git hash:** 8f397e04e5ce87fb5ff99d6659ba959c22ea926c
